backend/main/shared/dps.go

```diff
@@ -0,0 +1,213 @@
+package shared
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"strings"
+
+	"github.com/onflow/cadence"
+	dpsApi "github.com/onflow/flow-dps/api/dps"
+	"github.com/onflow/flow-dps/codec/zbor"
+	"github.com/onflow/flow-dps/service/invoker"
+	"github.com/onflow/flow-go-sdk"
+	_flow "github.com/onflow/flow-go/model/flow"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var (
+	DummyBalance = FTBalanceResponse{
+		PrimaryAccountBalance:   100,
+		SecondaryAccountBalance: 100,
+		StakingBalance:          100,
+		BlockHeight:             0,
+	}
+)
+
+type DpsAdapter struct {
+	Config  FlowConfig
+	Invoker *invoker.Invoker
+	Context context.Context
+	URL     string
+	Env     string
+}
+
+func NewDpsClient(flowEnv string) *DpsAdapter {
+	// Initialize adapter
+	adapter := DpsAdapter{}
+	adapter.Context = context.Background()
+	adapter.Env = flowEnv
+
+	// parse flow.json
+	path := "./flow.json"
+
+	content, err := ioutil.ReadFile(path)
+
+	if err != nil {
+		log.Fatal().Msgf("Error when opening file: %+v.", err)
+	}
+
+	var config FlowConfig
+	err = json.Unmarshal(content, &config)
+	if err != nil {
+		log.Fatal().Msgf("Error parsing flow.json: %+v.", err)
+	}
+
+	adapter.Config = config
+	adapter.URL = config.DpsNetworks[adapter.Env]
+
+	// For emulator, skip invoker instantiation
+	if adapter.URL == "" {
+		return &adapter
+	}
+
+	// Initialize the DPS API client.
+	codec := zbor.NewCodec()
+
+	conn, err := grpc.DialContext(
+		adapter.Context,
+		adapter.URL,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.FailOnNonTempDialError(true), // DialContext will fail immediately if passed bad connection string
```

Comment:
do we require the `insecure` package for some special reason? Seems as if `credentials` grpc could be used to pass in credentials..